iothub/device/src/IotHubBaseClient.cs

```diff
@@ -194,7 +199,11 @@ public async Task SendTelemetryBatchAsync(IEnumerable<TelemetryMessage> messages
         /// Calling this API more than once will result in the callback set last overwriting any previously set callback.
         /// A method callback can be unset by setting <paramref name="messageCallback"/> to null.
         /// </remarks>
-        /// <param name="messageCallback">The callback to be invoked when a cloud-to-device message is received by the client.</param>
+        /// <param name="messageCallback">
+        /// The callback to be invoked when a cloud-to-device message is received by the client.
+        /// This user-supplied callback is awaited by the SDK. All of requests will be processed as they arrive.
```

Comment:
What do you think about putting this explanation in the remarks section for this method definition? Having a whole lot of text in the param name field might be overwhelming to some people.